Uplift/Views/DropdownView.swift

```diff
@@ -0,0 +1,190 @@
+        self.headerView = headerView
+        self.headerView.isUserInteractionEnabled = true
+        self.headerView.translatesAutoresizingMaskIntoConstraints = false
+        addSubview(self.headerView)
+        
+        self.headerViewHeight = headerViewHeight
+        currentHeight = headerViewHeight
+        
+        expandCollapseDropdownGesture = UITapGestureRecognizer(target: self, action: #selector(self.expandCollapseDropdown(sender:) ))
+        self.headerView.addGestureRecognizer(expandCollapseDropdownGesture)
+        
+        self.contentView = contentView
+        self.contentView.translatesAutoresizingMaskIntoConstraints = false
+        addSubview(self.contentView)
+        
+        currentHeight = headerView.bounds.height
+        
+        self.contentViewHeight = contentViewHeight
+        
+        self.halfDropdownEnabled = halfDropdownEnabled
+        
+        if let halfView = halfExpandView {
+            self.halfExpandView = halfView
+            self.halfExpandView!.translatesAutoresizingMaskIntoConstraints = false
+            self.halfExpandView!.isHidden = true
+            self.halfExpandView?.isUserInteractionEnabled = true
+            addSubview(self.halfExpandView!)
+            expandFromHalfDropdownGesture = UITapGestureRecognizer(target: self, action: #selector(self.expandFromHalfDropdown(sender:) ))
+            halfView.addGestureRecognizer(expandFromHalfDropdownGesture)
+        }
+        
+        self.halfExpandViewHeight = halfExpandViewHeight
+        
+        if let halfView = halfCollapseView {
+            self.halfCollapseView = halfView
+            self.halfCollapseView!.isHidden = true
+            self.halfCollapseView!.isUserInteractionEnabled = true
+            self.halfCollapseView!.translatesAutoresizingMaskIntoConstraints = false
+            addSubview(self.halfCollapseView!)
+            collapseHalfDropdownGesture = UITapGestureRecognizer(target: self, action: #selector(self.collapseHalfDropdown(sender:) ))
+            halfView.addGestureRecognizer(collapseHalfDropdownGesture)
+        }
+        
+        self.halfCollapseViewHeight = halfCollapseViewHeight
+        
+        self.halfHeight = halfHeight
+        
+        setupConstraints()
+    }
+    
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    func setupConstraints() {
+        NSLayoutConstraint.activate([
+            headerView.topAnchor.constraint(equalTo: topAnchor),
+            headerView.leadingAnchor.constraint(equalTo: leadingAnchor),
+            headerView.trailingAnchor.constraint(equalTo: trailingAnchor),
+            headerView.heightAnchor.constraint(equalToConstant: headerViewHeight),
+            contentView.topAnchor.constraint(equalTo: headerView.bottomAnchor),
+            contentView.leadingAnchor.constraint(equalTo: leadingAnchor),
+            contentView.trailingAnchor.constraint(equalTo: trailingAnchor)
+        ])
```

Comment:
why aren't we using `SnapKit`?

Reply:
I needed to maintain references to `contentViewBottomConstraint` and `contentViewHeightConstraint` since their constants are changed in order to change the dropdown view's height, so I used `NSLayoutConstraint`s for those. I used `NSLayoutConstraint`s for the other constraints for the sake of consistency.

Reply:
You can hold on to references of snapkit constraints as well, if you do `.constraint` on the end of setting the constraint, 
ex: `self.topConstraint = make.top.equalTo(superview).constraint`